contributors/design-proposals/node-allocatable.md

```diff
@@ -89,61 +73,204 @@ The flag will be specified as a serialized `ResourceList`, with resources define
 --kube-reserved=cpu=500m,memory=5Mi
 ```
 
-Initially we will only support CPU and memory, but will eventually support more resources. See
-[#16889](https://github.com/kubernetes/kubernetes/pull/16889) for disk accounting.
-
-If KubeReserved is not set it defaults to a sane value (TBD) calculated from machine capacity. If it
-is explicitly set to 0 (along with `SystemReserved`), then `Allocatable == Capacity`, and the system
-behavior is equivalent to the 1.1 behavior with scheduling based on Capacity.
+Initially we will only support CPU and memory, but will eventually support more resources like [local storage](#phase-3) and io proportional weights to improve node reliability.
 
 #### System-Reserved
 
 In the initial implementation, `SystemReserved` will be functionally equivalent to
 [`KubeReserved`](#kube-reserved), but with a different semantic meaning. While KubeReserved
 designates resources set aside for kubernetes components, SystemReserved designates resources set
 aside for non-kubernetes components (currently this is reported as all the processes lumped
-together in the `/system` raw container).
+together in the `/system` raw container on non-systemd nodes).
+
+## Recommended Cgroups Setup
+
+Following is the recommended cgroup configuration for Kubernetes nodes.
+All OS system daemons are expected to be placed under a top level `SystemReserved` cgroup.
+`Kubelet` and `Container Runtime` are expected to be placed under `KubeReserved` cgroup.
+The reason for recommending placing the `Container Runtime` under `KubeReserved` is as follows:
+
+1. A container runtime on Kubernetes nodes is not expected to be used outside of the Kubelet.
+1. It's resource consumption is tied to the number of pods running on a node.
+
+Note that the hierarchy below recommends having dedicated cgroups for kubelet and the runtime to individally track their usage.
+
+/ (Cgroup Root)
+.
++..systemreserved or system.slice (`SystemReserved` enforced here *optionally* by kubelet)
+.        .    .tasks(sshd,udev,etc)
+.
+.
++..kubereserved or kube.slice (`KubeReserved` enforced here *optionally* by kubelet)
+.	 .
+.	 +..kubelet
+.	 .   .tasks(kubelet)
+.        .
+.	 +..runtime
+.	     .tasks(docker-engine, containerd)
+.	 
+.
++..kubepods or kubepods.slice (Node Allocatable enforced here by Kubelet)
+.	 .
+.	 +..PodGuaranteed
+.	 .	  .
+.	 .	  +..Container1
+.	 .	  .        .tasks(container processes)
+.	 .	  .
+.	 .        +..PodOverhead
+.	 .        .        .tasks(per-pod processes)
```

Comment:
seems like this would take docker-level support for putting all pause container processes in a single cgroup, if that is indeed what this `PodOverhead` cgroup is for.